From 13fd90a7a6290937157ce54a095f4b7ec5c75a09 Mon Sep 17 00:00:00 2001
From: Mohsen Dolaty <mohsen.dolaty@amd.com>
Date: Tue, 13 Sep 2022 10:11:17 -0500
Subject: [PATCH 1/1] linux-aspeed: Enable APML Over I2C

Add code to Device Table to enable APML Over I2C if needed

Signed-off-by: Mohsen Dolaty <mohsen.dolaty@amd.com>
---
 arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts     | 32 ++++++----
 arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts   | 58 +++++++++++-------
 arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts     | 31 ++++++----
 arch/arm/boot/dts/aspeed-bmc-amd-titanite.dts | 59 +++++++++++--------
 4 files changed, 112 insertions(+), 68 deletions(-)

diff --git a/arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts b/arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts
index aef95085a33a..99ac9b81a88e 100644
--- a/arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts
+++ b/arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts
@@ -733,6 +733,26 @@ rcd_1_5: rcd@5d,2C000000005 {
 	};
 };

+#ifdef ENABLE_I2C_APML
+
+&i2c2 {
+	// P0 APML - i2c10
+	status = "okay";
+	bus-frequency = <400000>;
+
+	sbrmi@3c {
+	compatible = "amd,sbrmi";
+		reg = <0x3c>;
+	};
+
+	sbtsi@4c {
+		compatible = "amd,sbtsi";
+		reg = <0x4c>;
+	};
+};
+
+#else
+
 &i3c4 {
 	// P0 APML
 	status = "okay";
@@ -755,18 +775,6 @@ sbrmi_p0_1: sbrmi@3c,22400000002 {
 	};
 };

-#ifdef ENABLE_I2C_APML
-
-&i2c2 {
-	// P0 APML - i2c10
-	status = "okay";
-	bus-frequency = <400000>;
-
-	sbtsi@4c {
-		compatible = "amd,sbtsi";
-		reg = <0x4c>;
-	};
-};
 #endif  // ENABLE_I2C_APML

 &espi_ctrl {
diff --git a/arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts b/arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts
index d62e34a4765d..4ca85b2b669d 100644
--- a/arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts
+++ b/arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts
@@ -1024,6 +1024,42 @@ rcd_3_5: rcd@5d,2C000000005 {
 	};
 };

+#ifdef ENABLE_I2C_APML
+
+&i2c2 {
+	// P0 APML - i2c10
+	status = "okay";
+	bus-frequency = <400000>;
+
+	sbrmi@3c {
+		compatible = "amd,sbrmi";
+		reg = <0x3c>;
+	};
+
+	sbtsi@4c {
+		compatible = "amd,sbtsi";
+		reg = <0x4c>;
+	};
+};
+
+&i2c3 {
+	// P1 APML - i2c11
+	status = "okay";
+	bus-frequency = <400000>;
+
+	sbrmi@38 {
+		compatible = "amd,sbrmi";
+		reg = <0x38>;
+	};
+
+	sbtsi@48 {
+		compatible = "amd,sbtsi";
+		reg = <0x48>;
+	};
+};
+
+#else
+
 &i3c4 {
 	// P0 APML
 	status = "okay";
@@ -1068,28 +1104,6 @@ sbrmi_p1_1: sbrmi@38,22400000002 {
 	};
  };

-#ifdef ENABLE_I2C_APML
-&i2c2 {
-	// P0 APML - i2c10
-	status = "okay";
-	bus-frequency = <400000>;
-
-	sbtsi@4c {
-		compatible = "amd,sbtsi";
-		reg = <0x4c>;
-	};
-};
-
-&i2c3 {
-	// P1 APML - i2c11
-	status = "okay";
-	bus-frequency = <400000>;
-
-	sbtsi@48 {
-		compatible = "amd,sbtsi";
-		reg = <0x48>;
-	};
-};
 #endif  // ENABLE_I2C_APML

 &espi_ctrl {
diff --git a/arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts b/arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts
index 6917a85c11d7..5e1b0bcb30bb 100644
--- a/arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts
+++ b/arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts
@@ -786,6 +786,26 @@ rcd_3_5: rcd@5d,2C000000005 {
         };
 };

+#ifdef ENABLE_I2C_APML
+
+&i2c2 {
+	// P0 APML - i2c10
+	status = "okay";
+	bus-frequency = <400000>;
+
+	sbrmi@3c {
+		compatible = "amd,sbrmi";
+		reg = <0x3c>;
+	};
+
+	sbtsi@4c {
+		compatible = "amd,sbtsi";
+		reg = <0x4c>;
+	};
+};
+
+#else
+
 &i3c4 {
 	// P0 APML
 	status = "okay";
@@ -808,17 +828,6 @@ sbrmi_p0_1: sbrmi@3c,22400000002 {
 	};
 };

-#ifdef ENABLE_I2C_APML
-
-&i2c2 {
-	// P0 APML
-	status = "okay";
-
-	sbtsi@4c {
-		compatible = "amd,sbtsi";
-		reg = <0x4c>;
-	};
-};
 #endif  // OPENBMC_I2C_ENABLE

 &espi_ctrl {
diff --git a/arch/arm/boot/dts/aspeed-bmc-amd-titanite.dts b/arch/arm/boot/dts/aspeed-bmc-amd-titanite.dts
index d61f7bbbc94e..77b3066d7305 100644
--- a/arch/arm/boot/dts/aspeed-bmc-amd-titanite.dts
+++ b/arch/arm/boot/dts/aspeed-bmc-amd-titanite.dts
@@ -521,6 +521,42 @@ rcd_3_5: rcd@5d,2C000000005 {
 	};
 };

+#ifdef ENABLE_I2C_APML
+
+&i2c2 {
+	// P0 APML - i2c10
+	status = "okay";
+	bus-frequency = <400000>;
+
+	sbrmi@3c {
+		compatible = "amd,sbrmi";
+		reg = <0x3c>;
+	};
+
+	sbtsi@4c {
+		compatible = "amd,sbtsi";
+		reg = <0x4c>;
+	};
+};
+
+&i2c3 {
+	// P1 APML - i2c11
+	status = "okay";
+	bus-frequency = <400000>;
+
+	sbrmi@38 {
+		compatible = "amd,sbrmi";
+		reg = <0x38>;
+	};
+
+	sbtsi@48 {
+		compatible = "amd,sbtsi";
+		reg = <0x48>;
+	};
+};
+
+#else
+
 &i3c4 {
 	// P0 APML
 	status = "okay";
@@ -565,29 +601,6 @@ sbrmi_p1_1: sbrmi@38,22400000002 {
 	};
 };

-#ifdef ENABLE_I2C_APML
-
-&i2c2 {
-	// P0 APML - i2c10
-	status = "okay";
-	bus-frequency = <400000>;
-
-	sbtsi@4c {
-		compatible = "amd,sbtsi";
-		reg = <0x4c>;
-	};
-};
-
-&i2c3 {
-	// P1 APML - i2c11
-	status = "okay";
-	bus-frequency = <400000>;
-
-	sbtsi@48 {
-		compatible = "amd,sbtsi";
-		reg = <0x48>;
-	};
-};
 #endif  // ENABLE_I2C_APML

 // Misc Regulators
--
2.25.1
